xbmc/filesystem/ImageFile.cpp

```diff
@@ -45,7 +45,12 @@ bool CImageFile::Exists(const CURL& url)
   std::string cachedFile =
       CServiceBroker::GetTextureCache()->CheckCachedImage(url.Get(), needsRecaching);
   if (!cachedFile.empty())
-    return CFile::Exists(cachedFile, false);
+  {
+    if (CFile::Exists(cachedFile, false))
+      return true;
+    else
+      CServiceBroker::GetTextureCache()->ClearCachedImage(url.Get());
```

Comment:
The commit description is good and describes the logic, but it might not be apparent to code readers why we're mutating the cache in an Exists() function. Maybe document the reason in the code too?